src/charm.py

```diff
@@ -157,6 +158,21 @@ def _on_get_issued_certificates(self, event: ActionEvent) -> None:
         results = {"certificates": [certificate.to_json() for certificate in certificates]}
         event.set_results(results)
 
+    def _on_rotate_private_key(self, event: ActionEvent):
+        """Handle the rotate-private-key action.
+
+        Args:
+            event (ActionEvent): Juju event
+        """
+        if not self.unit.is_leader():
+            event.fail("This action can only be run on the leader unit.")
+            return
+        self.tls_certificates.revoke_all_certificates()
+        self._clean_up_juju_secret(EXPIRING_CA_CERTIFICATES_SECRET_LABEL)
```

Comment:
This is somewhat related but if the secret does not exist, it will not be logged. We should probably add a log part of `_clean_up_juju_secret` to let the user know that this secret did not exist.

Reply:
Done